model/version/version.go

```diff
@@ -51,6 +52,18 @@ type Version struct {
 	AuthorID string `bson:"author_id,omitempty" json:"author_id,omitempty"`
 }
 
+func (v *Version) LastSuccessful() (*Version, error) {
+	versions, err := Find(BySuccessfulBeforeRevision(v.Identifier, v.RevisionOrderNumber).Sort(
```

Comment:
This is not covered by indexes
Also, why not just use FindOne?

Reply:
The identifier_1_r_1_order_1 makes this efficient enough. Changed to a findone